### PR TITLE
Documentation: Fix AsyncTestClient examples

### PR DESCRIPTION
**Pull request by odiseo0**
This PR addresses: issue #1099 

# PR Checklist

- [ ] Have you followed the guidelines in `CONTRIBUTING.md`?
- [ ] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

### CODE REVIEW

File: docs/usage/testing.rst [hunk 1]
```diff
@@ -61,7 +61,8 @@ We would then test it using the test client like so:
             from my_app.main import app
 
 
-            def test_health_check():
+            @pytest.mark.asyncio
```
Comment (provinzkraut) [inline docs/usage/testing.rst:64]:
```suggestion
```
This is not necessary since we are using the "auto" mode of pytest-asyncio

Comment (odiseo0) [reply docs/usage/testing.rst:64]: Thanks for this, I was wondering why the other tests doesn't are flagged with that line. Sorry

**Comment by provinzkraut** [comment]
@odiseo0 Looks like the examples just need to be tested. You should be able to extract them as-is to `docs/examples`. If you put them in a file with a `test_`-prefix, pytest should pick them up.

File: docs/examples/tests/test_using_async_client.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from typing import Any, Dict
```
Comment (provinzkraut) [inline docs/examples/tests/test_using_async_client.py:1]: This is the same code that's already in [`docs/examles/testing/get_session_data_async.py`](https://github.com/starlite-api/starlite/blob/main/docs/examples/testing/get_session_data_async.py). What is the purpose of copying this?

**Comment by provinzkraut** [comment]
It's still missing the `.. literalinclude::`. Right now the code is just duplicated, which means the actual example isn't being tested. It should look like the ["using sessions"](https://github.com/starlite-api/starlite/pull/1112/files#diff-3aa1294d54e6e53ff45010020eff67c6864b409bb496b0d5f537bf211b464643L165) examples.

**Comment by odiseo0** [comment]
> It's still missing the `.. literalinclude::`. Right now the code is just duplicated, which means the actual example isn't being tested. It should look like the ["using sessions"](https://github.com/starlite-api/starlite/pull/1112/files#diff-3aa1294d54e6e53ff45010020eff67c6864b409bb496b0d5f537bf211b464643L165) examples.

Thank you, will take a look after work!

File: docs/examples/testing/test_health_check_sync.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import pytest
```
Comment (provinzkraut) [inline docs/examples/testing/test_health_check_sync.py:1]: Okay so you've got this tested now, but it's still just duplicated from the docs. What remains is to replace the `.. code-block::` in the documentation file this is taken from with a `.. literalinclude:: /examples/testing/test_health_check_sync.py`. This applies to `/examples/testing/test_health_check_async.py` as well.

**Review by provinzkraut — CHANGES_REQUESTED** [review]

**Comment by odiseo0** [comment]
Don't review yet, I have reorganize some things

**Comment by provinzkraut** [comment]
> Don't review yet, I have to reorganize some things

I converted it back into a draft :+1:

**Comment by odiseo0** [comment]
I did not have to reorganize anything more, ready for review

**Review by Jarmos-san — APPROVED** [review]
LGTM....can you fix the merge conflicts though? I think its because the docs file structure has changed. The `usage` directory is now under `docs/lib` instead.

**Comment by Goldziher** [comment]
ping. @odiseo0 - why is this taking so long? can you please get this finished?

**Review by Jarmos-san — APPROVED** [review]
LGTM...Great work on this PR, can merge it. :+1:

File: docs/lib/usage/testing.rst [hunk 4]
```diff
@@ -261,7 +245,7 @@ Another helper is the :class:`RequestFactory <starlite.testing.RequestFactory>`
 you need to test logic that expects to receive a request object.
 
 For example, lets say we wanted to unit test a *guard* function in isolation, to which end we'll reuse the examples
-from the :doc:`route guards </lib/usage/security/guards>` documentation:
+from the :doc:`route guards </usage/security/guards>` documentation:
```
Comment (provinzkraut) [inline docs/lib/usage/testing.rst:248]:
```suggestion
from the :doc:`route guards </lib/usage/security/guards>` documentation:
```

File: docs/lib/usage/testing.rst [hunk 5]
```diff
@@ -116,39 +116,23 @@ We would then be able to rewrite our test like so:
...
-                async with test_client as client:
-                    response = await client.get("/health-check")
-                    assert response.status_code == HTTP_200_OK
-                    assert response.text == "healthy"
 
 Using sessions
 ++++++++++++++
 
-If you are using :ref:`session middleware <lib/usage/middleware/builtin-middleware:session middleware>` for session persistence
+If you are using :ref:`session middleware <usage/middleware/builtin-middleware:session middleware>` for session persistence
```
Comment (provinzkraut) [inline docs/lib/usage/testing.rst:135]:
```suggestion
If you are using :ref:`session middleware <lib/usage/middleware/builtin-middleware:session middleware>` for session persistence
```

**Review by provinzkraut — CHANGES_REQUESTED** [review]

**Pull request merged**